core/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -206,8 +207,13 @@ public PluginsService(Settings settings, Path modulesDirectory, Path pluginsDire
                 }
                 Class moduleClass = method.getParameterTypes()[0];
                 if (!Module.class.isAssignableFrom(moduleClass)) {
-                    logger.warn("Plugin: {} implementing onModule by the type is not of Module type {}", pluginEntry.v1().getName(), moduleClass);
-                    continue;
+                    String extraMessage = "";
```

Comment:
> I could also just remove it and rely on the developer to read the breaking changes docs

This is my preference, but I trust your judgment. You have an LGTM from me whatever you decide.